lib/internal/bootstrap/pre_execution.js

```diff
@@ -168,6 +169,53 @@ function setupSignalHandlers() {
   process.on('removeListener', stopListeningIfSignal);
 }
 
+
+function setupStacktracePrinterOnSigint() {
+  if (!getOptionValue('--trace-sigint')) {
+    return;
+  }
+  const { SigintWatchdog, SignalFlags } = require('internal/watchdog');
+
+  const watchdog = new SigintWatchdog();
+  watchdog.oninterrupt = function oninterrupt(flag) {
+    if (process.listenerCount('SIGINT') > 0) {
+      // Simply ignore this watchdog interrupt to prevent duplicate
+      // handling of SIGINT.
+      return;
+    }
+    const console = require('console');
+
+    // Keyboard interruption may not be captured by uncaughtException
+    // listener.
+    // eslint-disable-next-line no-restricted-syntax
+    const err = new Error('Script execution was interrupted by `SIGINT`');
+    Object.defineProperty(err, 'name', {
+      value: 'KEYBOARD_INTERRUPT',
+      enumerable: false,
+      writable: true,
+      configurable: true
+    });
+    if (flag === SignalFlags.FromIdle) {
+      const stackTraceLimit = Error.stackTraceLimit;
+      Error.stackTraceLimit = 0;
```

Comment:
why are we setting this value to `0` here? Any particular reason?

Reply:
The stacktrace printed for SIGINT triggered during idle (i.e. uv-polling) isn't much related to the actual calling stack. So just disabling it.